docs/on-premises/03-user-guide/index.md

```diff
@@ -0,0 +1 @@
+# User Guide
```

Comment:
I only created the page so we could link to it and have a structure. I haven't found a built-in TOC, but with the sidebar it felt like unnecessary duplication.

The reason I haven't added content here is because I plan to merge the downstream RHEL user guide into this one in the next PR/step.